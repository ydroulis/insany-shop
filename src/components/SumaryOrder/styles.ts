"use client"

import styled from 'styled-components'

export const Container = styled.section`
    min-width: 352px;
    height: 700px;

    background-color: #fff;

    margin-top: 2.5rem;

    color: #41414D;

    padding: 1.25rem 1.5rem 1.5rem 1.5rem;

    position: relative;

    @media (min-width: 768px) {
        min-width: 282px;
    }

    @media (min-width: 1024px) {

    }

    @media (min-width: 1440px) {
        min-width: 352px;
    }
`

export const Title = styled.h2`
    font-size: 1.25rem;
    font-weight: 600;

    text-transform: uppercase;
    margin-bottom: 1.85rem;
`

export const Subtotal = styled.div`
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    font-size: 1rem;

    margin-bottom: 1rem;
`

export const Delivery = styled.div`
    width: 100%;    
    display: flex;
    justify-content: space-between;
    align-items: center;
    font-size: 1rem;

    margin-bottom: 1.6rem;
`

export const Divider = styled.div`
    width: 100%;
    height: 1px;
    background-color: #DCE2E5;

    margin-bottom: 0.75rem;
`

export const Total = styled.div`
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    font-size: 1rem;
    font-weight: 600;

    margin-bottom: 2.5rem;
`

export const Links = styled.ul`
    position: absolute;
    bottom: 24px;
    left: 24px;

    display: flex;
    flex-direction: column;
    gap: 12px;

    list-style: none;

    li a{
        color: #737380;
        text-decoration: underline;
        text-transform: uppercase;
    }
`