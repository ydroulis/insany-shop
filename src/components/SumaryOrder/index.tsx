import React from 'react';

import * as S from './styles';
import Button from '../Button';
import Link from 'next/link';

interface SumaryOrderProps {
    subtotal: number,
}

const SumaryOrder: React.FC<SumaryOrderProps> = ({ subtotal }) => {
    const delivery = 40;
    const total = subtotal + delivery;
    const deliveryFormatted = new Intl.NumberFormat('pt-BR', {
        style: 'currency',
        currency: 'BRL',
    }).format(delivery);
    const totlaFormatted = new Intl.NumberFormat('pt-BR', {
        style: 'currency',
        currency: 'BRL',
    }).format(total);
    const subtotalFormatted = new Intl.NumberFormat('pt-BR', {
        style: 'currency',
        currency: 'BRL',
    }).format(subtotal);

    return (
        <S.Container aria-labelledby="summary-title" role="region">
            <S.Title id="summary-title">Resumo do pedido</S.Title>

            <dl>
                <S.Subtotal>
                    <dt>Subtotal</dt>
                    <dd>{subtotalFormatted}</dd>
                </S.Subtotal>

                <S.Delivery>
                    <dt>Entrega</dt>
                    <dd>{deliveryFormatted}</dd>
                </S.Delivery>

                <S.Divider />

                <S.Total>
                    <dt>Total</dt>
                    <dd>
                        <strong aria-live="polite">{totlaFormatted}</strong>
                    </dd>
                </S.Total>
            </dl>

            <Button
                action={() => { }}
                type="checkout"
                ariaLabel="Finalizar a compra e ir para o checkout"
            >
                Finalizar compra
            </Button>

            <nav aria-label="Links de suporte">
                <S.Links>
                    <li>
                        <Link href="#">Ajuda</Link>
                    </li>
                    <li>
                        <Link href="#">Reembolso</Link>
                    </li>
                    <li>
                        <Link href="#">Entregas e Frete</Link>
                    </li>
                    <li>
                        <Link href="#">Trocas e Devolução</Link>
                    </li>
                </S.Links>
            </nav>
        </S.Container>
    );
}

export default SumaryOrder;