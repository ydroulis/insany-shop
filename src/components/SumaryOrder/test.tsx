import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import SumaryOrder from "./";

jest.mock("../Button", () => {
    const MockButton = ({
        action,
        ariaLabel,
        children,
    }: {
        action: () => void;
        ariaLabel?: string;
        children: React.ReactNode;
    }) => (
        <button onClick={action} aria-label={ariaLabel}>
            {children}
        </button>
    );
    MockButton.displayName = "Button";
    return MockButton;
});

describe("<SumaryOrder />", () => {
    const subtotal = 100;

    it("should render the title and be accessible via aria-labelledby", () => {
        render(<SumaryOrder subtotal={subtotal} />);
        const region = screen.getByRole("region", { name: /resumo do pedido/i });
        expect(region).toBeInTheDocument();

        const title = screen.getByRole("heading", { name: /resumo do pedido/i });
        expect(title).toBeInTheDocument();
    });

    it("should render subtotal, delivery and total with correct currency formatting", () => {
        render(<SumaryOrder subtotal={subtotal} />);

        expect(screen.getByText("Subtotal")).toBeInTheDocument();
        expect(screen.getByText("R$ 100,00")).toBeInTheDocument();

        expect(screen.getByText("Entrega")).toBeInTheDocument();
        expect(screen.getByText("R$ 40,00")).toBeInTheDocument();

        expect(screen.getByText("Total")).toBeInTheDocument();
        expect(screen.getByText("R$ 140,00")).toBeInTheDocument();
    });

    it("should render the checkout button with correct aria-label", () => {
        render(<SumaryOrder subtotal={subtotal} />);

        const button = screen.getByRole("button", {
            name: /finalizar a compra e ir para o checkout/i,
        });

        expect(button).toBeInTheDocument();
        fireEvent.click(button);
    });

    it("should render support links inside a navigation landmark", () => {
        render(<SumaryOrder subtotal={subtotal} />);
        const nav = screen.getByRole("navigation", { name: /links de suporte/i });
        expect(nav).toBeInTheDocument();

        const links = screen.getAllByRole("link");
        const expectedLinks = ["Ajuda", "Reembolso", "Entregas e Frete", "Trocas e Devolução"];

        expectedLinks.forEach((text) => {
            expect(
                screen.getByRole("link", { name: new RegExp(text, "i") })
            ).toBeInTheDocument();
        });

        expect(links).toHaveLength(expectedLinks.length);
    });

    it("should match snapshot", () => {
        const { container } = render(<SumaryOrder subtotal={subtotal} />);
        expect(container).toMatchSnapshot();
    });
});
