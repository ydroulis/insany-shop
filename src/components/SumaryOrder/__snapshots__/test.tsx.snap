// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<SumaryOrder /> should match snapshot 1`] = `
<div>
  <section
    aria-labelledby="summary-title"
    class="sc-bRKDuR cbuulU"
    role="region"
  >
    <h2
      class="sc-hvigdm cTwYRL"
      id="summary-title"
    >
      Resumo do pedido
    </h2>
    <dl>
      <div
        class="sc-fhHczv fwGnLo"
      >
        <dt>
          Subtotal
        </dt>
        <dd>
          R$ 100,00
        </dd>
      </div>
      <div
        class="sc-ggWZvA kFMIOj"
      >
        <dt>
          Entrega
        </dt>
        <dd>
          R$ 40,00
        </dd>
      </div>
      <div
        class="sc-dTvVRJ jglFJG"
      />
      <div
        class="sc-jwTyAe fyVbHG"
      >
        <dt>
          Total
        </dt>
        <dd>
          <strong
            aria-live="polite"
          >
            R$ 140,00
          </strong>
        </dd>
      </div>
    </dl>
    <button
      aria-label="Finalizar a compra e ir para o checkout"
    >
      Finalizar compra
    </button>
    <nav
      aria-label="Links de suporte"
    >
      <ul
        class="sc-hjsuWn cJXQoz"
      >
        <li>
          <a
            href="#"
          >
            Ajuda
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Reembolso
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Entregas e Frete
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Trocas e Devolução
          </a>
        </li>
      </ul>
    </nav>
  </section>
</div>
`;
